hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -313,57 +318,75 @@ commit ctx st utxo = do
       rejectMoreThanMainnetLimit networkId utxo
       Right $ commitTx networkId scriptRegistry headId ownParty utxo initial
  where
-  ChainContext{networkId, ownParty, ownVerificationKey, scriptRegistry} = ctx
+  ChainContext{networkId, ownParty, scriptRegistry} = ctx
+  InitialState{headId} = st
 
-  InitialState
-    { initialInitials
-    , seedTxIn
-    , headId
-    } = st
+-- | Construct a commit script transaction based on the 'InitialState'.
+--  This does look for "our initial output" to spend and check the given 'UTxO' to be
+-- compatible. Hence, this function does fail if already committed.
+draftCommitTxBody ::
+  ChainContext ->
+  InitialState ->
+  UTxO ->
+  [(TxIn, BuildTxWith BuildTx (Witness WitCtxTxIn))] ->
+  Either (PostTxError Tx) Tx
+draftCommitTxBody ctx st utxo scriptWitnesses = do
+  txBody <- case ownInitial ctx st of
+    Nothing ->
+      Left (CannotFindOwnInitial{knownUTxO = getKnownUTxO st})
+    Just initial -> do
+      rejectByronAddress utxo
+      rejectReferenceScripts utxo
+      rejectMoreThanMainnetLimit networkId utxo
```

Comment:
Must: not duplicate logic when it is not needed, but update the signature of `commit` above instead.

```
commit ::
  ChainContext ->
  InitialState ->
  UTxO' (TxOut CtxUTxO, Maybe (ScriptWitness WitCtxTxIn Era)) ->
  Either (PostTxError Tx) Tx
```

could work and should simplify the call site as well.